src/modules/AuthService.test.ts

```diff
@@ -0,0 +1,50 @@
+import * as AuthService from "src/modules/AuthService";
+import { User } from "src/modules/User";
+
+jest.mock("firebase", () => ({
+  initializeApp: jest.fn(() => {
+    const functions = require("firebase-functions-test");
+    const testEnv = functions();
+    const firebaseUser = testEnv.auth.exampleUserRecord();
+    return {
+      auth: () => {
+        return {
+          signInWithEmailAndPassword: () =>
+            Promise.resolve({ user: firebaseUser }),
```

Comment:
We also need to mock these functions as per: https://jestjs.io/docs/mock-functions#using-a-mock-function
So that we can check if the same was called in the respective wrapper with a check like:

```js
// The first argument of the first call to the function was 0
expect(signInWithEmialAndPassword.mock.calls[0][0]).toBe(0);
```

Reply:
function that lies inside another function can't be mocked as you mentioned.

Reply:
These two examples are different from our case. Can you try something like:

```js
const mockSignInWithEmailAndPassword = jest.fn(({email, password}) =>  Promise.resolve({ user: firebaseUser }));

jest.mock("firebase", () => ({
  ...
  return {
    auth: () => ({
        signInWithEmailAndPassword,
        ...
     })
  }
});

....

expect(mockSignInWithEmailAndPassword.mock.calls[0][0]).toBe("user@gmail.com");

Reply:
mockSignInWithEmailAndPassword is simple mock function it can't test `AuthService.ts` file.

Reply:
Can you put this on a codesandbox with the above approach or share some references

Reply:
example you have given doesn't work so don't see the point of putting this in sandbox

Reply:
The codesandbox would allow me to explain the mocking. Can you create one and post the link here

Reply:
https://codesandbox.io/s/hopeful-carson-dmpio?file=/src/index.tsx

Reply:
Jest is not fully supported in codesandbox: https://github.com/codesandbox/codesandbox-client/issues/513